bench/workgen/workgen.cpp

```diff
@@ -353,42 +359,97 @@ gen_random_table_name(char *name, workgen_random_state volatile *rand_state)
  * fails.
  */
 int
-WorkloadRunner::create_table(WT_SESSION *session, const std::string &config, const std::string &uri,
-  const std::string &mirror_uri, const bool is_base)
+WorkloadRunner::create_table(
+  WT_SESSION *session, const std::string &config, const std::string &uri, bool mirror_enabled)
 {
-    // Check if a table with this name already exists. Return if it does. Use a shared lock to
-    // read the dynamic table structure.
+    const std::string mirror_uri(mirror_enabled ? uri + _workload->options.mirror_suffix : "");
+
+    // Return if a table with the same name already exists. If mirror is enabled, make sure it does
+    // not exist either.
     ContextInternal *icontext = _workload->_context->_internal;
     {
+        // Use a shared lock to read the dynamic table structure.
         const std::shared_lock lock(*icontext->_dyn_mutex);
         if (icontext->_tint.count(uri) > 0 || icontext->_dyn_tint.count(uri) > 0)
             return EEXIST;
+        if (mirror_enabled &&
+          (icontext->_tint.count(mirror_uri) > 0 || icontext->_dyn_tint.count(mirror_uri) > 0))
+            return EEXIST;
     }
 
-    // Create the table.
-    WT_DECL_RET;
-    if ((ret = session->create(session, uri.c_str(), config.c_str())) != 0) {
-        if (ret != EBUSY)
-            THROW("Failed to create table '" << uri << "'.");
+    /*
+     * When mirror is enabled, create the mirror first and then the base. If we create the base
+     * first, threads may start working on the base while the mirror is not fully created.
+     *
+     * The config has to be updated with the following information:
+     *  - The name of the table's mirror if mirroring is enabled.
+     *  - If this table is a base table or a mirror.
+     *
+     * See below an example when creating the table 'a' with mirror enabled:
+     *  - Configuration expected for the mirror:
+     * app_metadata="workgen_dynamic_table=true,workgen_table_mirror=table:a,workgen_base_table=false"
+     *  - Configuration expected for the base:
+     * app_metadata="workgen_dynamic_table=true,workgen_table_mirror=table:a_mirror,workgen_base_table=true"
+     *
+     * Without mirror enabled:
+     * app_metadata="workgen_dynamic_table=true,workgen_base_table=true"
+     *
+     */
+    std::string mirror_config;
+    if (mirror_enabled) {
+        mirror_config = config + "," + MIRROR_TABLE_APP_METADATA + uri + "," +
+          BASE_TABLE_APP_METADATA + "false\"";
+        int ret = session->create(session, mirror_uri.c_str(), mirror_config.c_str());
```

Comment:
NP: Could this be simplified with WT_RET_MSG?

Reply:
I wanted to stick the `VERBOSE` macro that is used across the framework.

Reply:
I see, understood.